docs/source/examples/Detecting Events.mystnb

```diff
@@ -0,0 +1,386 @@
+k = Earth.k.to_value(u.km ** 3 / u.s ** 2)
+R_sec = Sun.R.to_value(u.km)
+R_pri = Earth.R.to_value(u.km)
+# Position vector of Sun wrt Solar System Barycenter
+r_sec_ssb = get_body_barycentric_posvel("Sun", orbit.epoch)[0]
+r_pri_ssb = get_body_barycentric_posvel("Earth", orbit.epoch)[0]
+
+r_sec = ((r_sec_ssb - r_pri_ssb).xyz << u.km).value
+
+rr = (rr << u.km).value
+vv = (vv << u.km / u.s).value
+
+eclipses = []  # List to store values of eclipse_function.
+for i in range(len(rr)):
+    r = rr[i]
+    v = vv[i]
+    eclipse = eclipse_function(k, np.hstack((r, v)), r_sec, R_sec, R_pri)
+    eclipses.append(eclipse)
+
+plt.xlabel("Time (s)")
+plt.ylabel("Eclipse function")
+plt.title("Eclipse function vs time")
+plt.plot(tofs[:len(rr)].to_value(u.s), eclipses);
+```
+
+<div class="alert alert-info">For simplicity, here we compute the position vector of the primary and the secondary body only once, at the orbit epoch. However, the eclipse events internally recompute the position vectors at each desired instant.</div>
+
+We could get some geometrical insights by plotting the orbit:
+
+```{code-cell} ipython3
+# Plot `Earth` at the instant of event occurence.
+Earth.plot(orbit.epoch+umbra_event.last_t, label="Earth position at event instance");
+
+plotter = StaticOrbitPlotter()
+
+plotter.plot(orbit)
+plotter.set_orbit_frame(orbit)
+
+# Convert satellite coordinates to a `CartesianRepresentation` object.
+coords = CartesianRepresentation(rr[:, 0] << u.km, rr[:, 1] << u.km, rr[:, 2] << u.km)
+plotter.plot_trajectory(coords, color="orange");
+```
+
+It seems our satellite is exiting the umbra region, as is evident from the orange colored trajectory!
+
+## Node Cross Event
+
+```{code-cell} ipython3
+r = [-3182930.668, 94242.56, -85767.257] << u.km
+v = [505.848, 942.781, 7435.922] << u.km / u.s
+orbit = Orbit.from_vectors(Earth, r, v)
+```
+
+```{code-cell} ipython3
+node_event = NodeCrossEvent(terminal=True)
+events = [node_event]
+
+tofs = [0.01, 0.1, 0.5, 0.8, 1, 3, 5, 6, 10, 11, 12, 13, 14, 15] << u.s
+rr, vv = cowell(
+    Earth.k,
+    orbit.r,
+    orbit.v,
+    tofs,
+    events=events,
+)
+
+print(f"The nodal cross time was {node_event.last_t} after the orbit's epoch")
+```
+
+The plot below shows us the variation of the z coordinate of the orbit's position vector with time:
+
+```{code-cell} ipython3
+z_coords = [r[-1].to_value(u.km) for r in rr]
+plt.xlabel("Time (s)")
+plt.ylabel("Z coordinate of the position vector")
+plt.title("Node Crossing")
+plt.plot(tofs[:len(rr)].to_value(u.s), z_coords);
+```
+
+We could do the same plotting done in `LatitudeCrossEvent` to check for equatorial crossings:
+
+```{code-cell} ipython3
+es = EarthSatellite(orbit, None)
+
+# Show the groundtrack plot from
+t_span = time_range(orbit.epoch - 1.5 * u.h, end=orbit.epoch + node_event.last_t)
+
+# Generate ground track plotting instance.
+gp = GroundtrackPlotter()
+gp.update_layout(title="Node Crossing")
+
+# Plot the above-defined earth satellite.
```

Comment:
Use `###` if this is a sub-section of previous example.

Reply:
I am slightly confused by this. All the `#`'s except for the title are essentially code comments and not headers, isn't it?

Reply:
My fault! I thought this was a section title, sorry 😓